src/inspector/models/text/textsettingsmodel.cpp

```diff
@@ -100,55 +103,110 @@ void TextSettingsModel::requestElements()
+        Pid::PLACEMENT,
+        Pid::TEXT_SCRIPT_ALIGN
+    };
+
+    loadProperties(propertyIdSet);
+}
+
+void TextSettingsModel::loadProperties(const PropertyIdSet& propertyIdSet)
+{
+    if (muse::contains(propertyIdSet, Pid::FONT_FACE)) {
+        loadPropertyItem(m_fontFamily, [](const QVariant& elementPropertyValue) -> QVariant {
+            return elementPropertyValue.toString() == mu::engraving::TextBase::UNDEFINED_FONT_FAMILY
+                   ? QVariant() : elementPropertyValue.toString();
+        });
+
+        m_fontFamily->setIsEnabled(true);
+    }
 
-    loadPropertyItem(m_fontStyle, [](const QVariant& elementPropertyValue) -> QVariant {
-        return elementPropertyValue.toInt() == static_cast<int>(mu::engraving::FontStyle::Undefined)
-               ? QVariant() : elementPropertyValue.toInt();
-    });
+    if (muse::contains(propertyIdSet, Pid::FONT_STYLE)) {
+        loadPropertyItem(m_fontStyle, [](const QVariant& elementPropertyValue) -> QVariant {
+            return elementPropertyValue.toInt() == static_cast<int>(mu::engraving::FontStyle::Undefined)
+                   ? QVariant() : elementPropertyValue.toInt();
+        });
 
-    m_fontStyle->setIsEnabled(true);
+        m_fontStyle->setIsEnabled(true);
+    }
 
-    loadPropertyItem(m_fontSize, [](const QVariant& elementPropertyValue) -> QVariant {
-        return muse::RealIsEqual(elementPropertyValue.toDouble(), mu::engraving::TextBase::UNDEFINED_FONT_SIZE)
-               ? QVariant() : elementPropertyValue.toDouble();
-    });
+    if (muse::contains(propertyIdSet, Pid::FONT_SIZE)) {
+        loadPropertyItem(m_fontSize, [](const QVariant& elementPropertyValue) -> QVariant {
+            return muse::RealIsEqual(elementPropertyValue.toDouble(), mu::engraving::TextBase::UNDEFINED_FONT_SIZE)
+                   ? QVariant() : elementPropertyValue.toDouble();
+        });
 
-    m_fontSize->setIsEnabled(true);
+        m_fontSize->setIsEnabled(true);
+    }
 
-    loadPropertyItem(m_textLineSpacing, formatDoubleFunc);
+    if (muse::contains(propertyIdSet, Pid::TEXT_LINE_SPACING)) {
+        loadPropertyItem(m_textLineSpacing, formatDoubleFunc);
+    }
 
-    loadPropertyItem(m_horizontalAlignment, [](const QVariant& elementPropertyValue) -> QVariant {
-        QVariantList list = elementPropertyValue.toList();
-        return list.size() >= 2 ? list[0] : QVariant();
-    });
+    if (muse::contains(propertyIdSet, Pid::ALIGN)) {
+        loadPropertyItem(m_horizontalAlignment, [](const QVariant& elementPropertyValue) -> QVariant {
+            QVariantList list = elementPropertyValue.toList();
+            return list.size() >= 2 ? list[0] : QVariant();
+        });
+    }
 
-    loadPropertyItem(m_verticalAlignment, [](const QVariant& elementPropertyValue) -> QVariant {
-        QVariantList list = elementPropertyValue.toList();
-        return list.size() >= 2 ? list[1] : QVariant();
-    });
+    if (muse::contains(propertyIdSet, Pid::ALIGN)) {
```

Comment:
It looks like these two `if`s can be consolidated, or did you intentionally put them as two separate?

Reply:
No, no reason. You have a point. I combined them now. Thanks.